tests/spec/unit/templates/parser.spec.js

```diff
@@ -36,201 +37,150 @@ describe('Parser class', () => {
+                expect(actual).toBe(path.join(mockProjectPath, 'www'));
             });
+        });
 
-            const emitSpy = jasmine.createSpy('emit');
-            Parser.__set__('events', {
-                emit: emitSpy
+        describe('update_www method', () => {
+            it('should detect merges/electron file path and merge/update user source files into the platform staging dir', () => {
+                spyOn(fs, 'existsSync').and.returnValue(true);
+                spyOn(events, 'emit');
+
+                const mergeAndUpdateDirSpy = jasmine.createSpy('mergeAndUpdateDir');
+                Parser.__set__('FileUpdater', {
+                    mergeAndUpdateDir: mergeAndUpdateDirSpy
+                });
+
+                parser.update_www({
+                    root: 'mock',
+                    projectConfig: {
+                        path: path.join('mock', 'config.xml'),
+                        cdvNamespacePrefix: 'cdv'
+                    },
+                    locations: {
+                        www: path.join('mock', 'www')
+                    }
+                });
+
+                expect(events.emit).toHaveBeenCalledWith(
+                    'verbose',
+                    'Found "merges/electron" folder. Copying its contents into the electron project.'
+                );
+
+                const expectedSourceDirs = [
+                    'www',
+                    path.join('..', mockProjectPath, 'platform_www'),
+                    path.join('merges', 'electron')
+                ];
+                const expectedTargetDir = path.join('..', mockProjectPath, 'www');
+
+                expect(events.emit).toHaveBeenCalledWith(
+                    'verbose',
+                    `Merging and updating files from [${expectedSourceDirs.join(', ')}] to ${expectedTargetDir}`
+                );
+
+                expect(mergeAndUpdateDirSpy).toHaveBeenCalled();
             });
 
-            const mergeAndUpdateDirSpy = jasmine.createSpy('mergeAndUpdateDir');
-            Parser.__set__('FileUpdater', {
-                mergeAndUpdateDir: mergeAndUpdateDirSpy
+            it('should detect merges/electron file path and merge/update user source files into the platform staging dir', () => {
+                spyOn(fs, 'existsSync').and.returnValue(false);
+                spyOn(events, 'emit');
+
+                const mergeAndUpdateDirSpy = jasmine.createSpy('mergeAndUpdateDir');
+                Parser.__set__('FileUpdater', {
+                    mergeAndUpdateDir: mergeAndUpdateDirSpy
+                });
+
+                const cordovaProject = {
+                    root: 'mock',
+                    projectConfig: {
+                        path: path.join('mock', 'config.xml'),
+                        cdvNamespacePrefix: 'cdv'
+                    },
+                    locations: {
+                        www: path.join('mock', 'www')
+                    }
```

Comment:
I noticed an identical mock in this file. Maybe you could export it somewhere up above and re-use it.